app/controllers/jobs_controller.rb

```diff
@@ -2,6 +2,37 @@
 
 class JobsController < ApplicationController
   before_action :set_job, only: [:show]
+  before_action :authenticate_user!, only: %i(new create)
+
+  # GET /jobs/new
+  def new
+    # The organization needs to be built to display the nested form
+    @job = Job.new.tap(&:build_organization)
+  end
+
+  # POST /jobs
+  def create
+    @job = Job.new(permitted_attributes.merge(user_id: current_user.id))
+
+    if @job.save
+      flash[:notice] = "The job posting has been send for review. Thanks!"
+      redirect_to confirmation_job_url(@job)
+    else
+      render :new
+    end
+  end
+
+  # GET /jobs/1/confirmation
+  def confirmation
+    @job = Job.draft.where(user_id: current_user&.id).find_by_id(params[:id])
```

Comment:
This is nice scoping, but you can also use Pundit for authorizations.
```
@job = Job.find params[:id]
authorize(@job)
...
```

You'll have to define the `JobPolicy` class if it's not already defined and define the `confirmation?` method. You should also authorise in `new` and `create`.

Reply:
I never used Pundit. It seems really interesting! I'll take a look to the doc. Thanks for the heads up :)

Reply:
I've used it for years! If you want to pair program a bit or see some live examples, I can find some time!